aurora/line_broaden.py

```diff
@@ -0,0 +1,556 @@
+        )
+
+    # Output line shape and wavelength mesh
+    return {
+        'lams_profs_A': lams_profs_A,   # [AA], dim(trs,nlamb)
+        'theta': theta,                 # [1/AA], dim(trs,nlamb)
+        }
+
+# Calculates Natural broadening
+def _get_Natural(
+    # Settings
+    dphysics=None,
+    # ADF15 file
+    wave_A=None,
+    ):
+    '''
+    INPUTS: dphysics -- [dict], necessary physics information
+                NOTE: The expected form is keys for each central
+                wavelength of interst with its associated Einstein
+                coefficient as a float. This is so that a user won't
+                need to provide a data for every transition in the
+                ADF15 file.
+
+                !!! It is assumed you know, at least very closely,
+                the exact wavelength stored in your ADF15 file
+
+                i.e., If one wishes to explore the impact of Natural
+                broadening on the w, x, y, z lines for He-like Kr, the
+                input dictionary would look like --
+                dphysics = {
+                    '0.9454': 1.529e15, # [Hz], w line
+                    '0.9471': 9.327e10, # [Hz], x line
+                    '0.9518': 3.945e14, # [Hz], y line
+                    '0.9552': 5.715e9,  # [Hz], z line
+                    }
+                    ref: K.M. Aggarwal and F.P. Keenan, 2012 Phys. Scr. 86 035302
+
+            wave_A -- [list], dim(trs,), [AA], 
+                transition central wavelength
+
+    OUTPUTS: [dict], line shape
+                i) 'lam_profs_A' -- [AA], dim(trs,nlamb), 
+                        wavelength mesh for each transition
+                ii) 'theta' -- [1/AA], dim(trs, nlamb),
+                        line shape for each transition
+
+
+    '''
+
+    # Initializes output
+    lams_profs_A = np.zeros((len(wave_A), 101)) # [AA], dim(trs, nlamb)
+    theta = np.zeros((len(wave_A), 101)) # [1/AA], dim(trs, nlamb)
+
+    # Tolerance on wavelength match
+    tol = 1e-4 # [AA]
+
+    # Loop over transitions of interest
+    for lmb in dphysics.keys():
+        # Finds transitions of interst in ADF15 file
+        ind = np.where(
+            (wave_A >= float(lmb) -tol)
+            & (wave_A <= float(lmb)+tol)
+            )[0]
+
+        # Error check
+        if len(ind) == 0:
+            print('NO TRANSITION FOUND FOR LAMBDA= '+str(lmb))
+            continue
+
+        # FWHM
+        dnu = dphysics[lmb]/(2*np.pi) # [Hz]
+
+        # Loop over transitions
+        for ii in ind:
+            # Calculate Lorentzian shape
+            lams_profs_A[ii,:], theta[ii,:] = _calc_Lorentzian(
+                dnu = dnu,
+                wave_A=wave_A[ii],
+                )
+
+    # Output line shape and wavelength mesh
+    return {
+        'lams_profs_A': lams_profs_A,   # [AA], dim(trs,nlamb)
+        'theta': theta,                 # [1/AA], dim(trs,nlamb)
+        }
+
+# Calculates Instrumental broadening
+def _get_Instru(
```

Comment:
Generalize this to give the user a choose in the number of Gaussian/Lorentzian profiles that they've determined best models their instrumental broadening